tfupdate/option.go

```diff
@@ -48,13 +51,18 @@ func NewOption(updateType string, name string, version string, platforms []strin
 		regexps = append(regexps, r)
 	}
 
+	if sourceMatchType == "" {
```

Comment:
If sourceMatchType is invalid, other than `full` or `regex`, return with an error.
Since the option's default value is `full`, I prefer an explicit error to an implicit interpretation of an invalid value.